server/base/src/main/java/org/apache/accumulo/server/fs/VolumeChooserEnvironment.java

```diff
@@ -33,15 +35,29 @@
 
   private final ChooserScope scope;
   private final Table.ID tableId;
+  private final Text endRow;
 
   public VolumeChooserEnvironment(ChooserScope scope) {
     this.scope = Objects.requireNonNull(scope);
     this.tableId = null;
+    endRow = null;
   }
 
   public VolumeChooserEnvironment(Table.ID tableId) {
     this.scope = ChooserScope.TABLE;
     this.tableId = Objects.requireNonNull(tableId);
+    endRow = null;
+  }
+
+  public VolumeChooserEnvironment(Table.ID tableId, TabletId tabletId) {
+    this.scope = ChooserScope.TABLE;
+    this.tableId = Objects.requireNonNull(tableId);
+    this.endRow = tabletId.getEndRow();
```

Comment:
Could probably just store the tabletId and provide a getter for that, rather than a getter for the end row.